btree/btree.c

```diff
@@ -3917,14 +3939,17 @@ static int64_t btree_put_kv_tick(struct m0_sm_op *smop)
 		 * revert back the changes made on btree. Detailed
 		 * explination is provided at P_MAKESPACE stage.
 		 */
-		rec->r_flags = M0_BSC_SUCCESS;
-		int rc = bop->bo_cb.c_act(&bop->bo_cb, rec);
+		node_slot.s_rec.r_flags = M0_BSC_SUCCESS;
+		rc = bop->bo_cb.c_act(&bop->bo_cb, &node_slot.s_rec);
 		if (rc) {
 			/* handle if callback fail i.e undo make */
-			node_del(node_slot.s_node, node_slot.s_idx, bop->bo_tx);
-			node_done(&node_slot, bop->bo_tx, true);
-			node_seq_cnt_update(lev->l_node);
-			node_fix(lev->l_node, bop->bo_tx);
+			if (bop->bo_opc == M0_BO_PUT) {
```

Comment:
`if ( rc && bop->bo_opc == M0_BO_PUT)`

Reply:
line 3946 will get execute only if rec is non zero. no need to check for rc flag again

Reply:
I meant the if condition could be clubbed with `if(rc)`.

Reply:
I have not done that  as we need to execute line 3954-3956 if rc is non zero and  irrespective of bop->bo_opc code.
If we  (rc && bop->bo_opc == M0_BO_PUT)->execute error handling for PUT then again we need to check for rc, if it is non zero then we will execute line 3954-3956. I think it won't be readable.